ipfs/src/lib.rs

```diff
@@ -82,25 +166,26 @@ impl Handler<HttpStream> for IpfsHandler {
 				res.headers_mut().set(ContentLength(bytes.len() as u64));
 				res.headers_mut().set(ContentType(content_type));
 
-				Next::write()
 			},
 			NotFound(reason) => {
 				res.set_status(StatusCode::NotFound);
 
 				res.headers_mut().set(ContentLength(reason.len() as u64));
 				res.headers_mut().set(ContentType(mime!(Text/Plain)));
-
-				Next::write()
 			},
 			Bad(reason) => {
 				res.set_status(StatusCode::BadRequest);
 
 				res.headers_mut().set(ContentLength(reason.len() as u64));
 				res.headers_mut().set(ContentType(mime!(Text/Plain)));
-
-				Next::write()
 			}
 		}
+
+		if let Some(cors_header) = cors::get_cors_header(&self.cors_domains, &self.origin) {
+			res.headers_mut().set(cors_header);
```

Comment:
`Access-Control-Allow-Methods` and `Access-Control-Allow-Headers` are only necessary in preflight request, currently non-GET request get killed outright so there is no need for them. I've added `Vary`.